migrations.go

```diff
@@ -4,13 +4,78 @@ import (
 	"database/sql"
 	"encoding/json"
 	"fmt"
+	"sort"
 
 	"github.com/lib/pq/hstore"
 	"github.com/remind101/empire/internal/migrate"
 	"github.com/remind101/empire/pkg/constraints"
 	"github.com/remind101/empire/procfile"
 )
 
+const (
+	DefaultInstancePortPoolStart = 9000
+	DefaultInstancePortPoolEnd   = 10000
+)
+
+// DefaultSchema is the default Schema that will be used to migrate the database
+// if none is provided.
+var DefaultSchema = &Schema{
+	InstancePortPool: &InstancePortPool{
+		Start: DefaultInstancePortPoolStart,
+		End:   DefaultInstancePortPoolEnd,
+	},
+}
+
+type InstancePortPool struct {
+	Start, End uint
+}
+
+type Schema struct {
+	// For legacy ELB's (not ALB) Empire manages a pool of host ports to
+	// ensure that all applications have a unique port on the EC2 instance.
+	// This option specifies the beginning of that range.
+	InstancePortPool *InstancePortPool
+}
+
+// latestSchema returns the schema version that this version of Empire should be
+// using.
+func (s *Schema) latestSchema() int {
+	migrations := s.migrations()
+	return migrations[len(migrations)-1].ID
+}
+
+func (s *Schema) migrations() []migrate.Migration {
+	ports := migrate.Migration{
+		ID: 4,
+		Up: func(tx *sql.Tx) error {
+			const table = `CREATE TABLE ports (
+  id uuid NOT NULL DEFAULT uuid_generate_v4() primary key,
+  port integer,
+  app_id uuid references apps(id) ON DELETE SET NULL
```

Comment:
This actually gets updated in a later migration: https://github.com/remind101/empire/blob/master/migrations.go#L250-L251